test/integration/scheduler_perf/scheduler_perf_test.go

```diff
@@ -633,3 +634,13 @@ func getCustomVolumeFactory(pvTemplate *v1.PersistentVolume) func(id int) *v1.Pe
 		return pv
 	}
 }
+
+func isRetryableAPIError(err error) bool {
+	// These errors may indicate a transient error that we can retry in tests.
+	if apierrors.IsInternalError(err) || apierrors.IsTimeout(err) || apierrors.IsServerTimeout(err) {
```

Comment:
This looks better than my proposer here: https://github.com/kubernetes/kubernetes/pull/96704.
I wonder how do you judge those error are retriable?  Is there a doc to claim that? e.g. I saw that `StatusReasonServiceUnavailable` is also a retriable error from the code comments.
```
	// StatusReasonServiceUnavailable means that the request itself was valid,
	// but the requested service is unavailable at this time.
	// Retrying the request after some time might succeed.
	// Status code 503
	StatusReasonServiceUnavailable StatusReason = "ServiceUnavailable"
```

Obviously, not all the 5xx error is retriable.

Reply:
Timeouts are not generically retriable and neither are internal errors.  These errors can be produced, but the modifications may already have been taken, meaning that the post or put may already have happened.  I would avoid retrying these because we would not suggest that our clients generically do that.

Reply:
> I wonder how do you judge those error are retriable? Is there a doc to claim that?

@chendave I copied from a pre-existing snippet of `isRetryableAPIError()`.

@deads2k 's point is fair, esp. that retrying timed-out reqs that have already mutated API objs don't make sense. This PR only adds extra options to `CreatePod()` (not Update/Patch pods), if the timed-out req happens to have already created the pod, the following retry req would be safely caught by the error check of `AlreadyExists`, so generally IMO it would be fine without messing up the semantics.